cunumeric/array.py

```diff
@@ -3475,6 +3475,15 @@ def flip(self, axis=None) -> ndarray:
         return result
 
     def view(self, dtype=None, type=None):
+        import numpy
+
+        # It is valid to pass numpy.ndarray and subclasses thereof to this
+        # call, so we attempt to handle them here. If it's not a type a type
+        # that we recognize, assume it is a normal dtype.
+        if dtype is numpy.ndarray or dtype is ndarray:
```

Comment:
This code doesn't really handle the vanilla NumPy type correctly (`numpy.ndarray`). If the user asks for a conversion to `numpy.ndarray`, the current code returns a `cunumeric.ndarray`.

The original function is only guaranteed to work for subtypes of the array type, of which `cunumeric.ndarray` doesn't have any, so I think it's ok to fail when `dtype` is `numpy.ndarray`.

Reply:
I discussed this earlier with @magnatelee, and we thought silently returning a cunumeric array when asked for a numpy array makes sense, since we're trying to have cunumeric transparently drop in for numpy. I ran into this as some scipy library code calls `view(numpy.ndarray)`, for which I don't think cunumeric should error out if asked to do so.

Reply:
This should at last be documented in the docstring.

Reply:
We also have the ability to service this call properly, by actually returning a vanilla NumPy array that is a view over the same data as the cunumeric array, by calling `__array__()`, so this is a conscious choice that we are making here to break the contract of this function.

Why is scipy doing this? Are they trying to explicitly up-cast from an ndarray subclass to a base ndarray?

Reply:
Here's the snippet: https://github.com/scipy/scipy/blob/v1.8.1/scipy/sparse/_compressed.py#L473. They are handling any sort of array-like input to an element-wise multiply, and seem to be ensuring that they get a numpy array at least out of it.